### PR TITLE
Add the ability to manage row selection via useTable

### PR DESCRIPTION
**Pull request by wesleycole**
This will move all functionality and rendering of the checkboxes into the useTable hook so that the developer will only have to add the selectable prop to the useTable hook to make rows selectable.

**Main Story:** [ch61776](https://app.clubhouse.io/skyverge/story/61776/add-the-ability-to-manage-row-selection-via-usetable)

<a name="details"></a>

## Details

This PR will address changes to the following:

- [x] Components
- [x] Hooks

<a name="qa"></a>

## Acceptance Crtieria

- [x] When a user adds selectable: true to the useTable hook props the rows should be rendered with Kodiak UI checkboxes and be checkable.

<a name="API Changes"></a>

## API Changes

- [ ] **[Breaking]** This PR introduces breaking changes
- [ ] **[Documentation]** Documentation has been written about the API change

<a name="deployment"></a>

### Before merge to master

**Note**: _Code merged to master should be safe to automatically deploy to production as-is._

- [x] **[QA]** User-testing/quality assurance done
- [ ] **[Test]** All components and hooks should have a corresponding unit test
- [x] **[Storybook]** All components and hooks should have a corresponding Storybook story
- [ ] **[Documentation]** Documentation has been written or updated
- [ ] **[Version]** Version number has been updated

### CODE REVIEW

File: packages/table/src/useTable.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    getRowCheckbox,
+  } = useRowSelect({
+    selectable,
+    initialState: createRowState<T>({ data, state: false, initialSelectedIds }),
+    onSelect: (event, id) =>
+      setSelectedRows({
+        ...selectedRows,
+        [id]: event.target.checked,
+      }),
+    onSelectAll: event =>
```
Comment (alexanderchan) [inline packages/table/src/useTable.tsx:95]: Do we need a way to allow this to be overridden for a case when we want to only select the items on the page vs all of the items?

**Comment by wesleycole** [comment]
Yea, good point. Let's chat through that tomorrow. Will probably be worth handling that with the pagination implementation though.

File: packages/table/src/useTable.tsx (hunk 1 above)
Comment (wesleycole) [reply packages/table/src/useTable.tsx:95]: Yea, we will. Let's think through that scenario a bit more.

**Comment by alexanderchan** [comment]
@wesleycole looks like this got merged into the other branch instead of master.  We used to have the deletion bot but I think that got disabled, but there's a setting for the repo that we can turn on https://docs.github.com/en/github/administering-a-repository/managing-the-automatic-deletion-of-branches.  Think we'll just need a PR opened from the other branch that will have the changes.

**Pull request merged**